### PR TITLE
Keygen CLI

### PR DESCRIPTION
**Pull request by durch**
Rust utility for generating Coconut keypairs

### CODE REVIEW

File: keygen-cli/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    for keypair in keypairs {
+        debug!(
+            "Writing {:?} to keypairs/{}",
+            keypair.index,
+            run_id.to_string()
+        );
+        fs::write(
+            format!("keypairs/{}/{}", run_id.to_string(), keypair.index.unwrap()),
+            hex::encode(keypair.to_bytes()),
```
Comment (jstuczyn) [inline keygen-cli/src/main.rs:50]: The only thing I'm slightly unsure about is the encoding here - do we want to go with hex? I'm not saying whether it's good or bad, I don't know myself, perhaps @futurechimp might have some opinion, but so far everywhere in the system we've been using base58. Though I guess the only exception were mixnode (and gateway) keys that are implicitly converted to base64 when stored in .pem files.

Comment (durch) [reply keygen-cli/src/main.rs:50]: I've got no preference, I usually default to hex, but we can use base58 or 64, if we're already using them elsewhere :)

**Comment by jstuczyn** [comment]
Also, I've just realised, we should probably also introduce the same code (for serialization) for the Go codebase and make sure it's compatible.

**Comment by durch** [comment]
We only need the serialization for the keygen-cli, which is a rust binary, so we might be able to let is slide :D

**Pull request merged**